java/server/src/org/openqa/selenium/grid/distributor/Distributor.java

```diff
@@ -208,40 +234,32 @@ public CreateSessionResponse newSession(HttpRequest request)
         writeLock.unlock();
       }
 
-      CreateSessionResponse sessionResponse = selected
-        .orElseThrow(
-          () -> {
-            span.setAttribute("error", true);
-            SessionNotCreatedException
-              exception =
-              new SessionNotCreatedException(
-                "Unable to find provider for session: " + payload.stream()
-                  .map(Capabilities::toString).collect(Collectors.joining(", ")));
-            EXCEPTION.accept(attributeMap, exception);
-            attributeMap.put(AttributeKey.EXCEPTION_MESSAGE.getKey(),
-              EventAttribute.setValue(
-                "Unable to find provider for session: "
-                  + exception.getMessage()));
-            span.addEvent(AttributeKey.EXCEPTION_EVENT.getKey(), attributeMap);
-            return exception;
-          })
-        .get();
-
-      sessions.add(sessionResponse.getSession());
-
-      SessionId sessionId = sessionResponse.getSession().getId();
-      Capabilities caps = sessionResponse.getSession().getCapabilities();
-      String sessionUri = sessionResponse.getSession().getUri().toString();
-      SESSION_ID.accept(span, sessionId);
-      CAPABILITIES.accept(span, caps);
-      SESSION_ID_EVENT.accept(attributeMap, sessionId);
-      CAPABILITIES_EVENT.accept(attributeMap, caps);
-      span.setAttribute(AttributeKey.SESSION_URI.getKey(), sessionUri);
-      attributeMap.put(AttributeKey.SESSION_URI.getKey(), EventAttribute.setValue(sessionUri));
-
-      span.addEvent("Session created by the distributor", attributeMap);
-      return sessionResponse;
+      if (!selected.isPresent()) {
```

Comment:
For readability purposes, maybe it could be better to have the initial part of the `if` with the positive case (session created in this case), and the `else` with the negative one (session not created)

Reply:
Sure, will update that